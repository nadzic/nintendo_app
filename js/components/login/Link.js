import React, { Component } from 'react';
import { Text, StyleSheet, TouchableOpacity } from 'react-native';
import { COLORS } from 'nintendoapp/js/constants/colors';

/* eslint-disable global-require */

class Link extends Component {
  render() {
    const { text, testID, navFunction } = this.props;
    return (
      <TouchableOpacity
        onPress={navFunction}
        testID={testID}
      >
        <Text
          style={styles.contentText}
        >
          {text}
        </Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  contentText: {
    color: COLORS.WHITE,
    fontSize: 15,
    fontFamily: 'BrandonText-Medium',
  },
});

export default Link;
