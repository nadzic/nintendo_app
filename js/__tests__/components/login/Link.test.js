import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Link from 'nintendoapp/js/components/login/Link';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
  };

  const wrapper = shallow(<Link {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('Link component', () => {
  it('should renders correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
