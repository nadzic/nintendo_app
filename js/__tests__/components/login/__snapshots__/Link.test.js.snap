// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontFamily": "BrandonText-Medium",
        "fontSize": 15,
      }
    }
  />
</TouchableOpacity>
`;
